### PR TITLE
catppuccinifier-gui: init at 8.0.0

### PR DESCRIPTION
**Pull request by isabelroses**
## Description of changes

One of the issues with building catppuccinifier from source is that it uses `pngquant-bin` which needs to get remote binaries. Meaning its not able to properly reproduce using nix, so I used the native binaries instead.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Aleksanaa** [comment]
Wait so that's cli + gui?

**Comment by isabelroses** [comment]
> Wait so that's cli + gui?

just gui here.

**Comment by Aleksanaa** [comment]
> just gui here.

Then let's just call it catppuccinifier-gui, so it's easier to distinguish.

File: pkgs/by-name/ca/catppuccinifier/package.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        comment = "Apply catppuccin flavors to your wallpapers";
+      })
+    ];
+
+    meta = {
+      description = "Apply catppuccin flavors to your wallpapers";
+      homepage = "https://github.com/lighttigerXIV/catppuccinifier";
+      maintainers = with lib.maintainers; [isabelroses];
+      platforms = lib.platforms.linux;
+      mainProgram = "catppuccinifier";
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/catppuccinifier/package.nix:76]:
```suggestion
      mainProgram = "catppuccinifier";
      sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```

File: pkgs/by-name/ca/catppuccinifier/package.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      glib
+      dbus
+      openssl_3
+      librsvg
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      install -Dm555 installation-files/catppuccinifier-gui "$out/bin/catppuccinifier"
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/catppuccinifier/package.nix:55]: The executable can also remain it's original name.

File: pkgs/by-name/ca/catppuccinifier/package.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      install -Dm644 installation-files/catppuccinifier.png "$out/share/pixmaps/catppuccinifier.png"
+
+      runHook postInstall
+    '';
+
+    desktopItems = [
+      (makeDesktopItem {
+        desktopName = "catppuccinifier";
+        name = "catppuccinifier";
+        exec = "catppuccinifier";
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/catppuccinifier/package.nix:65]:
```suggestion
        exec = "catppuccinifier-gui";
```

**Review by Aleksanaa — COMMENTED** [review]
also you'd better reformat, this looks like alejandra

File: pkgs/by-name/ca/catppuccinifier/package.nix [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      icon = "catppuccinifier";
+      comment = "Apply catppuccin flavors to your wallpapers";
+    })
+  ];
+
+  meta = {
+    description = "Apply catppuccin flavors to your wallpapers";
+    homepage = "https://github.com/lighttigerXIV/catppuccinifier";
+    maintainers = with lib.maintainers; [ isabelroses ];
+    platforms = lib.platforms.linux;
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/catppuccinifier/package.nix:76]:
```suggestion
    platforms = [ "x86_64-linux" ];
```

File: pkgs/by-name/ca/catppuccinifier/package.nix [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      name = "catppuccinifier";
+      exec = "catppuccinifier-gui";
+      icon = "catppuccinifier";
+      comment = "Apply catppuccin flavors to your wallpapers";
+    })
+  ];
+
+  meta = {
+    description = "Apply catppuccin flavors to your wallpapers";
+    homepage = "https://github.com/lighttigerXIV/catppuccinifier";
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/catppuccinifier/package.nix:74]:
```suggestion
    homepage = "https://github.com/lighttigerXIV/catppuccinifier";
    licenses = lib.licenses.mit
```

**Review by Aleksanaa — COMMENTED** [review]
Also the file has to be moved into `pkgs/by-name/catppuccinifier-gui/package.nix`

**Comment by Aleksanaa** [comment]
Tested

**Pull request merged**